packages/serverless-component/__tests__/custom-inputs.test.js

```diff
@@ -431,4 +531,32 @@ describe("Custom inputs", () => {
       );
     });
   });
+
+  describe.each([
+    {
+      "some-invalid-page-route": { ttl: 100 }
+    },
+    {
+      "/api": { ttl: 100 }
+    },
+    { api: { ttl: 100 } },
+    { "api/test": { ttl: 100 } }
+  ])("Invalid cloudfront inputs", inputCloudfrontConfig => {
+    const fixturePath = path.join(__dirname, "./fixtures/generic-fixture");
+
+    beforeEach(async () => {
+      process.chdir(fixturePath);
+    });
+
+    it("throws an error", () => {
+      const component = createNextComponent();
+      expect(
+        // only deploy because the fixture will be cleaned up if
+        // build throws an error
+        component.deploy({
+          cloudfront: inputCloudfrontConfig
+        })
+      ).rejects.toThrow();
```

Comment:
The exception thrown here says  "childComponent is not a constructor". That doesn't seem right. Could you assert instead against a specific error message?